src/garage/sampler/base.py

```diff
@@ -1,43 +1,80 @@
+"""Base sampler class."""
```

Comment:
i want to start moving away from `base.py` everywhere. perhaps you can put this in `sampler.py`?